corehq/apps/reports/views.py

```diff
@@ -1242,6 +1244,12 @@ def dispatch(self, request, *args, **kwargs):
                           "Sorry, we couldn't find that case. If you think this "
                           "is a mistake please report an issue.")
             return HttpResponseRedirect(CaseListReport.get_url(domain=self.domain))
+        if not user_can_access_location_id(
+            self.domain,
+            request.couch_user,
+            self.case_instance.owner_id
+        ):
```

Comment:
It looks to be the report instance itself, so in this case - `self`.  That's kind of a weird dependency, IMO, but from a quick perusal, it looks like the only place it might give you trouble is [here](https://github.com/dimagi/commcare-hq/blob/ab6b8bb384539d479da6299452c407016d08f605/corehq/apps/reports/standard/cases/data_sources.py#L120), which expects the report to be a subclass of `ProjectReportParametersMixin`, which defines `individual`.  It looks like adding `individual = ''` [would work](https://github.com/dimagi/commcare-hq/blob/ab6b8bb384539d479da6299452c407016d08f605/corehq/apps/reports/standard/__init__.py#L110).  If not, you may just have to pull out that logic or duplicate it to figure out case ownership.